client_it_test.go

```diff
@@ -140,6 +143,64 @@ func TestClientMemberEvents(t *testing.T) {
 	})
 }
 
+func calcPartitionID(ss *serialization.Service, key interface{}) (int32, error) {
+	kd, err := ss.ToData(key)
+	if err != nil {
+		return 0, err
+	}
+	return murmur.HashToIndex(kd.PartitionHash(), 271), nil
+}
+
+func TestClientEventHandlingOrder(t *testing.T) {
+	// Create custom cluster, and client from it
+	cls := it.StartNewClusterWithOptions("event-order-test-cluster", 15701, it.MemberCount())
+	defer cls.Shutdown()
+	conf := cls.DefaultConfig()
+	ctx := context.Background()
+	c := it.MustValue(hz.StartNewClientWithConfig(ctx, conf)).(*hz.Client)
+	defer c.Shutdown(ctx)
+	ss := it.MustValue(serialization.NewService(&conf.Serialization)).(*serialization.Service)
+	// Create test map
+	m := it.MustValue(c.GetMap(ctx, "TestClientEventHandlingOrder")).(*hz.Map)
+	var lc hz.MapEntryListenerConfig
+	lc.NotifyEntryAdded(true)
+	var (
+		// have 271 partitions by default, starting from 1
+		partitionToEvent = make([][]int, 272)
+		// wait for all events to be processed
+		wg sync.WaitGroup
+		// access it with atomic package
+		count int32
+	)
+	for i := range partitionToEvent {
+		partitionToEvent[i] = make([]int, 10)
+	}
+	wg.Add(1)
+	it.MustValue(m.AddEntryListener(ctx, lc, func(event *hz.EntryNotified) {
```

Comment:
It would be nicer to extract the function to a variable.

Reply:
Fixed at https://github.com/hazelcast/hazelcast-go-client/pull/696/commits/db4dcd61d1d5f15c2887eabffc894e8c33884651